substrafl/dependency/path_management.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import shutil
+from abc import ABC
+from abc import abstractmethod
+from pathlib import PosixPath
+from typing import List
+from typing import Optional
+from typing import Set
+
+logger = logging.getLogger(__name__)
+
+TMP_SUBSTRAFL_PREFIX = "tmp_substrafl"
+EXCLUDED_PATHS_REGEX_DEFAULT = [
+    # Common data extensions
+    "*.csv",
+    "*.xls",
+    "*.xlsx",
+    # Python data extensiosn
+    "*.npy",
+    # Common image extensions
+    "*.png",
+    "*.jpg",
+    "*.jpeg",
```

Comment:
To match a more conventional exception, and to avoid weird side effect if a user want to load an image or a npy file, I would rather go for `data/*` instead of picking each data format.

Reply:
I think this is making a bigger assumption: that data-scientists use `data/*` to put their file

Reply:
From some discussion with Tanguy, it's not the case that all data is in the same folder. 
I think it's a bigger assumption to make to ignore everything in a `data` folder (that could contain, let's say, openers), than ignoring file formats. 
Obviously, we need to make it _very clear_  in the documentation, so that the user can add some files by hand if needed.

Reply:
I see for the opener... But I'm a bit worried about ignoring all files ending by *** cause it could lead to really weird behaviour if there is a png in a sub sub folder that the user forgot about...

Reply:
More generally, we could be inspired by the default gitignore for Python https://github.com/github/gitignore/blob/main/Python.gitignore

Reply:
What would be the weird behavior you're expecting? 

Which approach would you recommend? The first ones that are coming to mind:
- copy the file you linked and load it dynamically
- link (somehow) to the file in the other repo and load it dynamically
- embed the logic in our code (with potentially a script to do it)

Reply:
let's discuss that after the daily?